src/spiral_stc.cpp

```diff
@@ -237,19 +232,29 @@ namespace full_coverage_path_planner
         multiple_pass_counter--; // First point is already counted as visited
       }
 
-#ifdef DEBUG_PLOT
-      RCLCPP_INFO(rclcpp::get_logger("FullCoveragePathPlanner"), "Grid with path marked as visited is:");
-      gridNode_t SpiralStart = pathNodes.back();
-      printGrid(grid, visited, pathNodes, pathNodes.front(), pathNodes.back());
-#endif
+      gridNode_t lastNodeAstar = pathNodes.back();
 
       // Spiral fill from current position
       pathNodes = spiral(grid, pathNodes, visited);
 
-#ifdef DEBUG_PLOT
-      RCLCPP_INFO(rclcpp::get_logger("FullCoveragePathPlanner"), "Visited grid updated after spiral:");
-      printGrid(grid, visited, pathNodes, SpiralStart, pathNodes.back());
-#endif
+      // Need to extract only the spiral part for visualization
+      std::list<gridNode_t> pathNodes_copy = pathNodes; // Work with a copy so that splice() does not affect pathNodes
```

Comment:
The whole copy thing took me a while to grasp. Can we directly assign to `pathNodes_spiral` here? And process that variable until only the spiral is left.

Also nitpicking: mixing camelCase_underscore. Stick to underscores for all variables.

Reply:
Yes! I made it work with only pathNodes_spiral now.

The camelCase_underscore will be fixed further down the line. But a good one to keep in mind.